rust/src/verify.rs

```diff
@@ -170,6 +170,9 @@ fn ensure_left_most(spec: &ics23::InnerSpec, path: &[ics23::InnerOp]) -> Result<
     let pad = get_padding(spec, 0)?;
     for step in path {
         if !has_padding(step, &pad) {
+            if left_branches_are_empty(spec, step, 0)? {
```

Comment:
Maybe the same applies to Go, but looking at this I think it could be simplified:

``rust
if !has_padding(step, &pad) && !left_branches_are_empty(spec, step, 0)? {
    bail!("step not leftmost");
}
```